src/app/application/components/data-collection/DataCollection.vue

```diff
@@ -1,55 +1,86 @@
 <template>
-  <slot
-    v-if="items.length === 0"
-    name="empty"
-    :items="items"
+  <template
+    v-if="slots.item"
   >
-    <EmptyBlock v-if="props.empty" />
-  </slot>
-  <slot
+    <template
+      v-for="item in [props.items.find(props.predicate)]"
+      :key="item"
+    >
+      <slot
+        v-if="item"
+        name="item"
+        :item="item as T"
+      />
+      <slot
+        v-else
+        name="empty"
+        :items="items"
+      >
+        <EmptyBlock
+          v-if="props.empty"
+          :type="props.type"
+        />
+      </slot>
+    </template>
+  </template>
+  <template
     v-else
-    name="default"
-    :items="paginated"
-  />
-
-  <slot
-    v-if="props.page !== 0 && items.length > 0"
-    name="pagination"
-    :items="paginated"
   >
-    <KPagination
-      :total-count="items.length"
-      :current-page="props.page"
-      :initial-page-size="props.pageSize"
-      :page-sizes="[15, 30, 50, 75, 100]"
-      @page-change="({ page }: PaginationChangeEvent) => {
-        emit('change', {
-          page,
-          pageSize: props.pageSize,
-        })
-      }"
+    <slot
+      v-if="items.length === 0"
+      name="empty"
+      :items="items"
+    >
+      <EmptyBlock
+        v-if="props.empty"
+        :type="props.type"
+      />
+    </slot>
+    <slot
+      v-else
+      name="default"
+      :items="paginated"
     />
-  </slot>
+
+    <slot
+      v-if="props.page !== 0 && items.length > 0"
+      name="pagination"
+      :items="paginated"
+    >
+      <KPagination
+        :total-count="items.length"
+        :current-page="props.page"
+        :initial-page-size="props.pageSize"
+        :page-sizes="[15, 30, 50, 75, 100]"
+        @page-change="({ page }: PaginationChangeEvent) => {
+          emit('change', {
+            page,
+            pageSize: props.pageSize,
+          })
+        }"
+      />
+    </slot>
+  </template>
```

Comment:
A lot of the change here is just because I added an extra conditional/fork for the new `slots.item` which you'll see at the very top of this changeset.